tests/v2/validation/nodescaling/scaling_nodepools.go

```diff
@@ -37,6 +39,56 @@ func scalingRKE2K3SNodePools(t *testing.T, client *rancher.Client, clusterID str
 	pods.VerifyReadyDaemonsetPods(t, client, scaledClusterResp)
 }
 
+func scalingRKE2K3SCustomClusterNodes(t *testing.T, client *rancher.Client, clusterID string, nodeProvider string, nodeRoles machinepools.NodeRoles) {
+	rolesPerNode := []string{}
+	quantityPerPool := []int32{}
+	rolesPerPool := []string{}
+	for _, nodeRoles := range []machinepools.NodeRoles{nodeRoles} {
+		var finalRoleCommand string
+		if nodeRoles.ControlPlane {
+			finalRoleCommand += " --controlplane"
+		}
+		if nodeRoles.Etcd {
+			finalRoleCommand += " --etcd"
+		}
+		if nodeRoles.Worker {
+			finalRoleCommand += " --worker"
+		}
+		if nodeRoles.Windows {
+			finalRoleCommand += " --windows"
+		}
+		quantityPerPool = append(quantityPerPool, nodeRoles.Quantity)
+		rolesPerPool = append(rolesPerPool, finalRoleCommand)
+		for i := int32(0); i < nodeRoles.Quantity; i++ {
+			rolesPerNode = append(rolesPerNode, finalRoleCommand)
+		}
+	}
```

Comment:
this ties into my thinking around adding to extensions/provisioning, since this looks close to what we have in createCustomCluster function calls.